File: Java/RecursionPractice/src/calculations/Factorial.java
package calculations;

public class Factorial {

    int upToNumber;

    public int getValue() {
        return this.recursivelyCalculateValue(this.upToNumber);
    }

    private int recursivelyCalculateValue(int descendingNumber) {
        if (descendingNumber == 0 || descendingNumber == 1) {
            return 1;
        }
        return this.recursivelyCalculateValue(descendingNumber -1) * descendingNumber;
    }

    public Factorial(int upToNumber){
        this.upToNumber = upToNumber;
    }
}

File: Kotlin/RecursionPractice/src/calculations/Factorial.kt
package calculations

class Factorial(private val int: Int) {

    private fun recursivelyCalculate(descendingInt : Int): Int {
        if (descendingInt == 0 || descendingInt == 1) {
            return 1
        }
        return recursivelyCalculate(descendingInt - 1) * descendingInt
    }

    fun getValue(): Int {
        return recursivelyCalculate(int)
    }
}